services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/PortAntiflapSpec.groovy

```diff
@@ -0,0 +1,105 @@
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Issue
+import spock.lang.Narrative
+
+import java.util.concurrent.TimeUnit
+
+@Slf4j
+@Narrative("""
+Anti-flap feature is aimed to address the problem of blinking ports. Sometimes it happens that certain port on a switch
+begins to blink for a very short period of time. If it is a short-time behavior, in this case we don't want to init a
+port down event and reroute the flow. Otherwise we wait for 'antiflapWarmup' time and assign that blinking port a DOWN 
+status. Then, when it comes back up it should remain in a stable 'up' state for at least 'antiflapCooldown' seconds to
+actually become UP for the rest of the system.
+Initially, port is considered 'flapping' if it changes status quicker than once in 'antiflap.min' seconds (port will
+change status from UP to DOWN only after 'antiflap.min' in case of a single-time change of status)
+""")
+@Issue("https://github.com/telstra/open-kilda/issues/1729")
+class PortAntiflapSpec extends BaseSpecification {
+    @Value('${antiflap.min}')
+    int antiflapMin
+    @Value('${antiflap.warmup}')
+    int antiflapWarmup
+    @Value('${antiflap.cooldown}')
+    int antiflapCooldown
+    @Value('${discovery.interval}')
+    int discoveryInterval
+
+    @Autowired
+    IslUtils islUtils
+
+    def "Flapping port is brought down only after antiflap warmup and stable port is brought up only after cooldown \
+timeout"() {
+        given: "Switch, port and ISL related to that port"
+        def sw = topology.activeSwitches.first()
+        Isl isl = topology.islsForActiveSwitches.find { it.srcSwitch == sw || it.dstSwitch == sw }
+        assert isl
+        int islPort = isl.with { it.srcSwitch == sw ? it.srcPort : it.dstPort }
+
+        when: "ISL port begins to blink"
+        def blinker = new PortBlinker(northbound, sw, islPort)
+        def untilWarmupEnds = { blinker.timeStarted.time + antiflapWarmup * 1000 - new Date().time }
+        def untilCooldownEnds = { blinker.timeStopped.time + antiflapCooldown * 1000 - new Date().time }
+        blinker.start()
+
+        then: "Right before warmup timeout runs out the related ISL remains up"
+        sleep(untilWarmupEnds() - 100)
+        islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+
+        and: "After warmup timeout the related ISL goes down"
+        Wrappers.wait(untilWarmupEnds() / 1000.0 + WAIT_OFFSET / 2) {
+            islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "ISL remains down even after cooldown timeout"
+        TimeUnit.SECONDS.sleep(antiflapCooldown + 1)
+        islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+
+        when: "Port stops flapping, ending in UP state"
+        blinker.stop(true)
+
+        then: "Right before cooldown timeout runs out the ISL remains down"
+        sleep(untilCooldownEnds() - 100)
```

Comment:
100 ms safety delay may be a bit low if we run test against h/w environment

Reply:
The 'safety' here goes to the opposite direction. It is impossible for this event to occur earlier, but it may occur later due to internal Kilda operation time and network delays. The `untilCooldownEnds` is calculated locally and is a pretty accurate time. It of course will not be the same on the server, but it for sure won't be less than this.

Reply:
ok I see a problem here, need to investigate

Reply:
cannot test this properly due to #1790 issue which is caught by the test

Reply:
changed to -1000 for now